src/codegen/ObjectCodeFile.ts

```diff
@@ -6,7 +6,11 @@ import { getDocs } from "./getDocs.js";
 export const objectCodeGenerator = generatorFactory<IObjectDefinition>(
   async function() {
     const { typeName: { name }, docs } = this.def;
-    const fields = this.def.fields.map(f => `  ${f.fieldName}: ${this.getTypeForCode(f.type)};`)
+    const fields = this.def.fields.map(f =>
+      `  ${f.fieldName.includes("-") ? `"${f.fieldName}"` : f.fieldName}: ${
```

Comment:
is `-` the only character we have to worry about?

Reply:
Snake case is also valid, handling that